admin/admin.go

```diff
@@ -0,0 +1,70 @@
+package admin
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/gin-gonic/gin"
+)
+
+type AdminServer struct {
+	srv    *http.Server
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+
+	log log.Logger
+}
+
+func NewAdminServer(log log.Logger) *AdminServer {
+	return &AdminServer{log: log}
+}
+
+func (s *AdminServer) Start(ctx context.Context) error {
+	router := setupRouter()
+
+	s.srv = &http.Server{
+		Addr:    ":8420",
```

Comment:
should we make this port get automatically set based on which ports are available?

Reply:
I think it should be a stable port since devs may use it in CI to check if supersim is up!